CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+Developing DVOL
+---------------
+
+If you wish to work on DVOL, you'll first need [Go](http://www.golang.org) installed on your machine (version 1.6+ is *required*).
+
+For local dev first make sure Go is properly installed, including setting up a [GOPATH](http://golang.org/doc/code.html#GOPATH). You will also need to add `$GOPATH/bin` to your `$PATH`.
+
+
+Next, using [Git](https://git-scm.com/), clone this repository into `$GOPATH/src/github.com/ClusterHQ/dvol`. All the necessary Golang dependencies are vendored, however you will need to install some Python dependancies and Golang tools -
+
+```sh
+$ make bootstrap
+```
+
+To compile DVOL -
+
+```sh
+$ make build
+```
+
+To run the tests -
+
+```sh
+$ make test
+```
+
+### Dependencies
+
+Dvol stores its dependencies under `vendor/`, which [Go 1.6+ will automatically recognize and load](https://golang.org/cmd/go/#hdr-Vendor_Directories). We use [`godep`](https://github.com/tools/godep) to manage the vendored dependencies.
+
+If you're developing DVOL, there are a few tasks you might need to perform.
+
+#### Adding a dependency
+
+If you're adding a dependency. You'll need to vendor it in the same Pull Request as the code that depends on it. You should do this in a separate commit from your code, as makes PR review easier and Git history simpler to read in the future.
+
+Because godep captures new dependencies from the local `$GOPATH`, you first need to `godep restore` from the master branch to ensure that the only diff is your new dependency.
+
+Assuming your work is on a branch called `my-feature-branch`, the steps look like this:
+
+```bash
+# Get latest master branch's dependencies staged in local $GOPATH
+git checkout master
+git pull
+godep restore -v # flag is optional, enables verbose output
+
+# Capture the new dependency referenced from my-feature-branch
+git checkout my-feature-branch
+git rebase master
+godep save ./...
+
+# There should now be changes in `vendor/` with added files for your dependency,
+# and changes in Godeps/Godeps.json with metadata for your dependency.
+
+# Make a commit with your new dependencies added
+git add -A
```

Comment:
I think we might want to add more to the local .gitignore before encouraging people to do git add -A

specifically:
```
/cover.out
/dvol
/venv/
```